### PR TITLE
Add support for setting influxdb retention policy shard_duration

### PR DESCRIPTION
**Pull request by bbinet**
(See #52697 original PR for 2019.2)

This PR improves influx module and influxdb_retention_policy state:
- 0b8264d7877dc2cded7eeba63b567825baae29c0: fix unexpected overwrite of "default" param in influxdb.alter_retention_policy
- 8fac8a1c31833348c510b3cfc33ea9668952457c: add support for shard_duration in influx module
- bcef2397d47c2a3c9f879651b643557c411b344f: add support for shard_duration in influxdb_retention_policy state
- 6dbab0fcfac6441dcb1bbe27bdb4fcc653fbbcb0: fix some buggy edge cases in the convert_duration function and also support INF duration
- 649fadb48485808fb8f8399cc127b3a958799321: fix some false positive rp update when shard_duration='0s'

### CODE REVIEW

File: salt/modules/influxdbmod.py [hunk 1]
```diff
@@ -442,16 +455,17 @@ def create_retention_policy(database,
...
 
 
 def alter_retention_policy(database,
                            name,
-                           duration,
-                           replication,
-                           default=False,
+                           duration=None,
+                           replication=None,
+                           default=None,
```
Comment (Ch3LL) [inline salt/modules/influxdbmod.py]: Can you elaborate as to why having `default=False` is causing issues? I did notice on the api documentation that the default value should be None but just trying to understand the issue this is fixing

File: salt/modules/influxdbmod.py [hunk 2]
```diff
@@ -434,6 +435,18 @@ def create_retention_policy(database,
...
+
+        Durations such as 1h, 90m, 12h, 7d, and 4w, are all supported
+        and mean 1 hour, 90 minutes, 12 hours, 7 day, and 4 weeks,
+        respectively. Infinite retention is not supported.
+        As a workaround, specify a “1000w” duration to achieve an extremely
+        long shard group duration.
+        Defaults to “0s”, which is interpreted by the database to mean the
+        default value given the duration.
+        The minimum shard group duration is 1 hour.
+
```
Comment (Ch3LL) [inline salt/modules/influxdbmod.py]: will need to add `.. versionadded:: Neon` for each new argument added in this pr.

**Review by Ch3LL — CHANGES_REQUESTED** [review]
Thanks for splitting this out. I appreciate it . Unfortunately we will not be able to merge this until tests are added. Here are the docs about the test runner: https://docs.saltstack.com/en/latest/topics/tutorials/writing_tests.html

Let me know if i can help answer any other questions to help get you started. thanks

File: salt/modules/influxdbmod.py (hunk 1 above)
Comment (bbinet) [reply salt/modules/influxdbmod.py]: @Ch3LL You're right, after a few more tests, default=False is the same as default=None. I thought it had fixed my issue probably because of a side effect: the "default" value was already changed in my influxdb database.

**Comment by bbinet** [comment]
> Thanks for splitting this out. I appreciate it . Unfortunately we will not be able to merge this until tests are added. Here are the docs about the test runner: https://docs.saltstack.com/en/latest/topics/tutorials/writing_tests.html

@Ch3LL I understand, but sadly I won't have time right now to create new tests from scratch for influxdb module and states, I'm sorry (I could have done it if some tests would already exist for that module and states).
So let's close this PR. 
If someone wants to take on this task, please do ;-)
